### PR TITLE
Fixes invalid oci artifact if old component-descriptor media typ and nexus content-length bug

### PR DESCRIPTION
**Pull request by enrico-kaack-comp**
**What this PR does / why we need it**:
This PR fixes two smaller bugs:
- If a component-descriptor uses an old mediaType, the replicate_blobs logic now only compares for digest and not the whole object (including media type and digest).
- The nexus OCI registry does not return a Content-Length Header on head_manifest. Therefore, it is set to None in this case.

**Which issue(s) this PR fixes**:
Fixes #

**Special notes for your reviewer**:

**Release note**:

```bugfix developer
- component-descriptor patching now works correctly with old media type
- a missing content-length header from the nexus OCI registry is handled
```

### CODE REVIEW

File: oci/client.py [hunk 1]
```diff
@@ -527,10 +527,13 @@ def head_manifest(
         # XXX Docker-Content-Digest header may be absent or incorrect
         # -> it would be preferrable to retrieve the manifest and calculate the hash manually
 
+        if (size := headers.get('Content-Length', None)):
```
Comment (AndreasBurger) [inline oci/client.py:530]:
```suggestion
        if size := headers.get('Content-Length'):
```

File: oci/__init__.py [hunk 2]
```diff
@@ -138,9 +138,11 @@ def replicate_blobs(
 
     Note that the uploaded artifact must be finalised after the upload by a "manifest-put".
     '''
+    blob_overwrites = {k.digest:v for (k,v) in blob_overwrites.items()}
+
     def replicate_blob(blob: om.OciBlobRef) -> om.OciBlobRef:
-        if (blob_overwrite_bytes := blob_overwrites.get(blob)):
-            logger.info(f'overwriting {blob=}')
+        if (blob_overwrite_bytes := blob_overwrites.get(blob.digest)):
```
Comment (AndreasBurger) [inline oci/__init__.py:144]:
```suggestion
        if blob_overwrite_bytes := blob_overwrites.get(blob.digest):
```

File: oci/__init__.py [hunk 3]
```diff
@@ -138,9 +138,11 @@ def replicate_blobs(
 
     Note that the uploaded artifact must be finalised after the upload by a "manifest-put".
     '''
+    blob_overwrites = {k.digest:v for (k,v) in blob_overwrites.items()}
```
Comment (AndreasBurger) [inline oci/__init__.py:141]:
```suggestion
    blob_overwrites = {k.digest:v for k,v in blob_overwrites.items()}
```

**Review by AndreasBurger — APPROVED** [review]
Left a few suggestions if you want to save some parentheses. Feel free to disregard

**Review by ccwienk — APPROVED** [review]
lgtm - please squash

**Pull request merged**